src/Domain.cpp

```diff
@@ -740,24 +731,16 @@ unsigned long Domain::getglobalNumMolecules() {
 void Domain::setglobalNumMolecules(unsigned long glnummol) { _globalNumMolecules = glnummol; }
 
 void Domain::updateglobalNumMolecules(ParticleContainer* particleContainer, DomainDecompBase* domainDecomp) {
-		bool bNumPrtlsChangedGlobal = false;
-#ifdef ENABLE_MPI
-		MPI_Allreduce(&_bNumPrtlsChanged, &bNumPrtlsChangedGlobal, 1, MPI_CXX_BOOL, MPI_LOR, MPI_COMM_WORLD);
-#else
-		bNumPrtlsChangedGlobal = _bNumPrtlsChanged;
-#endif
-	if (bNumPrtlsChangedGlobal) {
-		CommVar<uint64_t> numMolecules;
-		numMolecules.local = particleContainer->getNumberOfParticles(ParticleIterator::ONLY_INNER_AND_BOUNDARY);
-		domainDecomp->collCommInit(1);
-		domainDecomp->collCommAppendUnsLong(numMolecules.local);
-		domainDecomp->collCommAllreduceSum();
-		numMolecules.global = domainDecomp->collCommGetUnsLong();
-		domainDecomp->collCommFinalize();
-		this->setglobalNumMolecules(numMolecules.global);
-		_bNumPrtlsChanged = false;
-		global_log->debug() << _localRank << " Updated global number of particles to N_new = " << this->getglobalNumMolecules() << std::endl;
-	}
+	unsigned long oldNum = _globalNumMolecules;
+	CommVar<uint64_t> numMolecules;
+	numMolecules.local = particleContainer->getNumberOfParticles(ParticleIterator::ONLY_INNER_AND_BOUNDARY);
+	domainDecomp->collCommInit(1);
+	domainDecomp->collCommAppendUnsLong(numMolecules.local);
+	domainDecomp->collCommAllreduceSum();
+	numMolecules.global = domainDecomp->collCommGetUnsLong();
+	domainDecomp->collCommFinalize();
+	this->setglobalNumMolecules(numMolecules.global);
+	std::cout << _localRank << " Updated global number of particles from " << oldNum << " to N_new = " << _globalNumMolecules << " ( " << numMolecules.global << " , " << numMolecules.local << " ) " << std::endl;
```

Comment:
Somehow, this is does not result in the correct number of particles. When calling the very same in the ResultWriter.cpp, the correct number is calculated. Strange...

Reply:
Problem solved...